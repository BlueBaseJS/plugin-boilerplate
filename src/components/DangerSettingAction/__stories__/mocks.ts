// tslint:disable: max-line-length object-literal-sort-keys
import gql from 'graphql-tag';

export const DeletePlaceMutation = gql`
	mutation DeletePlaceMutation($id: ID!) {
		deletePlace(id: $id)
	}
`;

export const networkError = {
	request: {
		query: DeletePlaceMutation,
		variables: {
			id: '123',
		},
	},
	error: Error('Boom!'),
};

export const success = [
	{
		request: {
			query: DeletePlaceMutation,
			variables: {
				id: '123',
			},
		},
		result: {
			data: {},
		},
	},
];
