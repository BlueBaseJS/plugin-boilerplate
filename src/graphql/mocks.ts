export * from './mutations/mocks';
export * from './queries/mocks';
