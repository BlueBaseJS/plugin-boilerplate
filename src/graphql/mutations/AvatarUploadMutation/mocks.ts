import { AvatarUpdateMutation } from './AvatarUploadMutation';
import faker from '@faker-js/faker';

export const AvatarUpdateMutationMocks = {
	success: {
		request: {
			query: AvatarUpdateMutation,
			variables: {
				data: {
					id: '123',
					image: 'dummy',
				},
			},
		},
		result: {
			data: {
				uploadBase64Image: {
					link: faker.image.avatar(),
				},
			},
		},
	},
};
