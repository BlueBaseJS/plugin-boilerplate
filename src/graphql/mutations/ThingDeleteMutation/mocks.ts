import { ThingDeleteMutation } from './ThingDeleteMutation';

const id = '123';

export const ThingDeleteMutationMocks = {
	error: {
		error: Error('Network Error!'),
		request: {
			query: ThingDeleteMutation,
			variables: {
				id,
			},
		},
	},

	success: {
		request: {
			query: ThingDeleteMutation,
			variables: {
				id,
			},
		},
		result: {
			data: { deleteThing: {} },
		},
	},
};
