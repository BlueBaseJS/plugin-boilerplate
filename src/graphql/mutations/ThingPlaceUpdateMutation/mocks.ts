import { ThingPlaceUpdateMutation } from './ThingPlaceUpdateMutation';

const id = '123';

export const ThingPlaceUpdateMutationMocks = {
	error: {
		error: Error('Network Error!'),
		request: {
			query: ThingPlaceUpdateMutation,
			variables: {
				id,
			},
		},
	},

	success: {
		request: {
			__typename: 'Thing',
			query: ThingPlaceUpdateMutation,
			variables: {
				clientMutationId: 'test-clientMutationId',
				data: {
					id,
					place: 'test-place-id-2',
				},
			},
		},
		result: {
			__typename: 'Thing',
			data: {
				clientMutationId: 'test-clientMutationId',
				updateThing: {
					thing: {
						__typeName: 'Thing',
						id: '123',
						name: 'test-name',
						place: {
							__typeName: 'Place',
							avatar: 'test-avatar',
							id: 'test-place-id-2',
							name: 'Second Place',
						},
					},
				},
			},
		},
	},
};
