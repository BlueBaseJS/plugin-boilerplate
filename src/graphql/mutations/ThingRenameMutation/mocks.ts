import { ThingRenameMutation } from './ThingRenameMutation';

const id = '123';

export const ThingRenameMutationMocks = {
	error: {
		error: Error('Network Error!'),
		request: {
			query: ThingRenameMutation,
			variables: {
				id,
			},
		},
	},

	success: {
		request: {
			query: ThingRenameMutation,
			variables: {
				clientMutationId: 'test-clientMutationId',
				data: {
					id,
					name: 'test-name1',
				},
			},
		},
		result: {
			data: {
				updateThing: {
					clientMutationId: 'test-clientMutationId',
					thing: {
						__typename: 'Thing',
						name: 'test-name1',
					},
				},
			},
		},
	},
};
