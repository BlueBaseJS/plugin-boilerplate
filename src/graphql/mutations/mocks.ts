export * from './ThingDeleteMutation/mocks';
export * from './ThingRenameMutation/mocks';
export * from './ThingPlaceUpdateMutation/mocks';
