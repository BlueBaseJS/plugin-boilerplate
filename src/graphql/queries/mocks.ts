export * from './ThingNodeQuery/mocks';
export * from './ThingPlaceQuery/mocks';
export * from './ThingProductQuery/mocks';
