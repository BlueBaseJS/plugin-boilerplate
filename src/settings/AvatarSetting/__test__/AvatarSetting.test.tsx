import 'cross-fetch/polyfill';

import Apollo from '@bluebase/plugin-apollo';
import { BlueBaseApp } from '@bluebase/core';
import BlueBasePluginResponsiveGrid from '@bluebase/plugin-responsive-grid';
import BlueEastClientPluginUI from '@blueeast/client-plugin-ui';
import MUI from '@bluebase/plugin-material-ui';
import { MockedProvider } from '@apollo/client/testing';
import PLuginUi from '@mevris/client-plugin-ui';
import PermissionUI from '@blueeast/client-plugin-permissions-ui';
import Plugin from '../../../index';
import React from 'react';
import { ThingNodeQueryMocks } from '../../../graphql/mocks';
import { ThingsAvatarSettings } from '../';
import { mount } from 'enzyme';
import { waitForElement } from 'enzyme-async-helpers';

const { success } = ThingNodeQueryMocks;
jest.mock('expo', () => {});
jest.mock('expo-location', () => ({ getCurrentPositionAsync: jest.fn() }));

describe('Avatar Setting', () => {
	it('Avatar Setting', async () => {
		// mount component

		const wrapper = mount(
			<BlueBaseApp
				plugins={[
					Plugin,
					MUI,
					Apollo,
					BlueBasePluginResponsiveGrid,
					PLuginUi,
					PermissionUI,
					BlueEastClientPluginUI,
				]}
			>
				<MockedProvider mocks={[success]} addTypename={true}>
					<ThingsAvatarSettings thingId="123" />
				</MockedProvider>
			</BlueBaseApp>
		);

		await waitForElement(wrapper, ThingsAvatarSettings);
		expect(wrapper.html()).toMatchSnapshot(); // 1st snapshot
		wrapper.setProps({ orderDetail: { source: false } });
		// check your snapshot with new props :)
		expect(wrapper.html()).toMatchSnapshot(); // 2nd snapshot
		const LoadingComponent: any = wrapper
			.find('StatefulComponent')
			.first()
			.prop('loadingComponent');
		LoadingComponent();
		wrapper.unmount();
	});
});
