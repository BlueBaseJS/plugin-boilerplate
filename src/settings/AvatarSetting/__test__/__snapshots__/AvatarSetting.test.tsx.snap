// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Setting Avatar Setting 1`] = `"<rncappearanceprovider style=\\"flex: 1;\\"><view style=\\"flex: 1;\\"><view style=\\"align-items: center; flex-grow: 1; justify-content: center; padding: 16px;\\"><text>TypeError</text><text>client_plugin_permissions_ui_1.usePermissions is not a function or its return value is not iterable</text></view></view></rncappearanceprovider>"`;

exports[`Avatar Setting Avatar Setting 2`] = `"<rncappearanceprovider style=\\"flex: 1;\\"><view style=\\"flex: 1;\\"><view style=\\"align-items: center; flex-grow: 1; justify-content: center; padding: 16px;\\"><text>TypeError</text><text>client_plugin_permissions_ui_1.usePermissions is not a function or its return value is not iterable</text></view></view></rncappearanceprovider>"`;
